Derpy.Tests/Utils/TumblrClientTest.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using Derpy.Utils;
+using Derpy.Utils.Tumblr;
+using Moq;
+using Xunit;
+
+namespace Derpy.Tests.Utils
+{
+    public class TumblrClientTest
+    {
+        private readonly TumblrClient _client;
+        private readonly Mock<IWebClient> _webClient = new Mock<IWebClient>();
+
+        public TumblrClientTest()
+        {
+            var keyProvider = new Mock<IKeyProvider>();
+            keyProvider.Setup(key => key.TumblrApiKey).Returns("test-api-key");
+            _client = new TumblrClient(_webClient.Object, keyProvider.Object);
+        }
+
+        [Fact]
+        public async void Test_FullResponse()
+        {
+            _webClient.Setup(webClient => webClient.GetAsync(UriFor("test-blog", null)))
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    Content = new StreamContent(LoadJsonResponse("full"))
+                });
+
+            var urls = await _client.GetAllPostUrlsAsync("test-blog");
+
+            Assert.Equal(3, urls.Length);
+            Assert.Equal(@"https://test-blog.tumblr.com/post/3507845453", urls[0]);
```

Comment:
What do the `@` mean here?

Reply:
These can be removed, must've left them in accidentally. An `@` defines a verbatim string literal, i.e. if you're writing out a Windows path you don't need to double escape your backslashes.